src/Components/BooksInputs.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
+
+const BooksInputs = () => {
+  const dispatch = useDispatch();
+
+  return (
+    <div>
+      <form onSubmit={(e) => {
+        e.preventDefault();
+        const title = e.target.elements.title.value;
+        const author = e.target.elements.author.value;
+        const payload = { title, author, id: uuidv4() };
+        dispatch(addBook(payload));
+        e.target.reset();
+      }}
+      >
+        <input type="text" placeholder="Book title.." id="title" required />
+        <input type="text" placeholder="Book author..." id="author" required />
+        <button type="submit">Add Book</button>
+      </form>
+    </div>
```

Comment:
- [ ] Your inputs should controlled. That means your input values should be changed using local state with the `onChange` event listener. You can visit [this article](https://blog.logrocket.com/controlled-vs-uncontrolled-components-in-react/) for reference

Reply:
> * [ ]  Your inputs should controlled. That means your input values should be changed using local state with the `onChange` event listener. You can visit [this article](https://blog.logrocket.com/controlled-vs-uncontrolled-components-in-react/) for reference

Hi, @julie-ify 
thanks for your valuable feedback, but I want to say that I read the project requirements Again, I couldn't find anything there that said (the inputs should be controlled). for reference below is the project requirement. 
 thanks a lot.

https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components_v2.md

Project requirements
1- Connect your store to the main component of your React app.
2 -Define a state slice which will contain your books
// Initial state:
[
  {
    "item_id": "item1",
    "title": "The Great Gatsby",
    "author": "John Smith",
    "category": "Fiction"
  },
  {
    "item_id": "item2",
    "title": "Anna Karenina",
    "author": "Leo Tolstoy",
    "category": "Fiction"
  },
  {
    "item_id": "item3",
    "title": "The Selfish Gene",
    "author": "Richard Dawkins",
    "category": "Nonfiction"
  }
]
3 -Display your books, received from the slice, in a reusable component
4 -Dispatch actions using useDispatch
5- Add a <Button> component, which includes: An event handler that adds a book to the books array (with attributes id, title and author)
7 -Add a <Button> component, which includes: An event handler that removes a book from the books array (by id)
8- Add a <Button> component, which includes: An event handler that check status (i.e add a functionality for "Check status" button click)